server/src/main/java/umm3601/Server.java

```diff
@@ -6,73 +6,120 @@
 import com.mongodb.ServerAddress;
 import com.mongodb.client.MongoClient;
 import com.mongodb.client.MongoClients;
-import com.mongodb.client.MongoDatabase;
 
 import org.bson.UuidRepresentation;
 
 import io.javalin.Javalin;
 import io.javalin.plugin.bundled.RouteOverviewPlugin;
 import io.javalin.http.InternalServerErrorResponse;
-import umm3601.user.UserController;
 
+/**
+ * The class used to configure and start a Javalin server.
+ */
 public class Server {
 
+  // The port that the server should run on.
   private static final int SERVER_PORT = 4567;
 
-  public static void main(String[] args) {
-
-    // Get the MongoDB address and database name from environment variables and
-    // if they aren't set, use the defaults of "localhost" and "dev".
-    String mongoAddr = System.getenv().getOrDefault("MONGO_ADDR", "localhost");
-    String databaseName = System.getenv().getOrDefault("MONGO_DB", "dev");
+  // The `mongoClient` field is used to access the MongoDB
+  private final MongoClient mongoClient;
+
+  // The `controllers` field is an array of all the `Controller` implementations
+  // for the server. This is used to add routes to the server.
+  private Controller[] controllers;
+
+  /**
+   * Construct a `Server` object that we'll use (via `startServer()`) to configure
+   * and start the server.
+   *
+   * @param mongoClient The MongoDB client object used to access to the database
+   * @param controllers The implementations of `Controller` used for this server
+   */
+  public Server(MongoClient mongoClient, Controller[] controllers) {
+    this.mongoClient = mongoClient;
+    // This is what is known as a "defensive copy". We make a copy of
+    // the array so that if the caller modifies the array after passing
+    // it in, we don't have to worry about it. If we didn't do this,
+    // the caller could modify the array after passing it in, and then
+    // we'd be using the modified array without realizing it.
+    this.controllers = Arrays.copyOf(controllers, controllers.length);
+  }
 
+  /**
+   * Setup the MongoDB database connection.
+   *
+   * This "wires up" the database using either system environment variables
+   * or default values. If you're running the server locally without any environment
+   * variables set, this will connect to the `dev` database running on your computer
+   * (`localhost`). If you're running the server on Digital Ocean using our setup
+   * script, this will connect to the production database running on server.
+   *
+   * This sets both the `mongoClient` and `database` fields
+   * so they can be used when setting up the Javalin server.
+ * @param mongoAddr The address of the MongoDB server
```

Comment:
For some reason, this is aligned differently from the other lines. Maybe it's just missing a space?

Reply:
No idea what happened there. This is why we should add `prettier` to the project (which there's a semi-old issue for).

I'd rather not block this merge over that if possible, but it obviously does need to get fixed. 😢